### ISSUE TITLE
Event UI not rendering

### ISSUE THREAD

**Issue by akrausesap** [kyma-project/kyma#2786]
**Description**
Dear Colleagues,
In Service Catalog the attached registration (Metadata) does not render.
![catalog](https://user-images.githubusercontent.com/40229965/52969429-a8288800-33b0-11e9-9cf4-7fb351ca706f.png)
[registration.json.zip](https://github.com/kyma-project/kyma/files/2876927/registration.json.zip)

The release in 0.7. For details  see attachment.
Cheers

Andreas

**Expected result**

Events get rendered in the UI

**Actual result**

Blank Page

**Steps to reproduce**

Register attached metadata, bind to namespace and browse catalog

**Troubleshooting**

N/A

**Comment by derberg** [comment]
@akrausesap Hey Andreas, thanks a lot for feedback. I need some details:
- is the `Doc` part working fine, it renders?
- could you check and put here the response from the GraphQL? if backend returns the spec or response is empty?

**Comment by akrausesap** [comment]
Hi @derberg,

Doc part is working fine.

```
{"data":{"clusterServiceClass":null,"serviceClass":{"name":"5a70282d-5c83-48b9-9fd2-8b92ab79c199","externalName":"sap-cloud-for-customer-c852b","displayName":"SAP Cloud for Customer","creationTimestamp":1550512882,"description":"SAP Cloud for Customer","longDescription":"SAP Cloud for Customer","documentationUrl":"","supportUrl":"","imageUrl":"","providerDisplayName":"my306768.vlab.sapbydesign.com","tags":[],"labels":{"connected-app":"c4c-vlab"},"content":{"description":"SAP Cloud for Customer OData APIs","displayName":"SAP Cloud for Customer OData APIs","docs":[{"source":"https://help.sap.com/viewer/1364b70b9cbb417ea5e2d80e966d4f49/latest/en-US","title":"SAP Cloud for Customer OData API","type":"Doc"},{"source":"https://help.sap.com/doc/d0f9ba822c08405da7d88174b304df84/LATEST/en-US/index.html","title":"SAP Cloud for Customer OData API 2.0 Reference","type":"Doc"},{"source":"https://github.com/SAP/C4CODATAAPIDEVGUIDE","title":"SAP Cloud for Customer OData API Developer’s Guide","type":"Doc"}],"tags":["SAP Cloud for Customer OData APIs"],"type":"SAP Cloud for Customer OData APIs"},"asyncApiSpec":{"asyncapi":"1.0.0","info":{"description":"SAP Cloud for Customer events","title":"SAP Cloud for Customer events","version":"v1"},"topics":{"Account.Root.Created.v1":{"subscribe":{"payload":{"properties":{"entity-id":{"description":"Account Root Object ID","example":{"entity-id":"00163E5FA59C1ED8B3B95BE1CB795280"},"properties":null,"required":true,"title":"Account Root Object ID","type":"string"},"root-entity-id":{"description":"Account Object ID","example":{"root-entity-id":"00163E5FA59C1ED8B3B9643A4B63D2AF"},"properties":null,"required":true,"title":"Account Object ID","type":"string"}},"required":["entity-id","root-entity-id"],"type":"object"},"summary":"Account.Root.Created.v1"}},"Account.Root.Deleted.v1":{"subscribe":{"payload":{"properties":{"entity-id":{"description":"Account Root Object ID","example":{"entity-id":"00163E5FA59C1ED8B3B95BE1CB795280"},"properties":null,"required":true,"title":"Account Root Object ID","type":"string"},"root-entity-id":{"description":"Account Object ID","example":{"root-entity-id":"00163E5FA59C1ED8B3B9643A4B63D2AF"},"properties":null,"required":true,"title":"Account Object ID","type":"string"}},"required":["entity-id","root-entity-id"],"type":"object"},"summary":"Account.Root.Deleted.v1"}},"Account.Root.Updated.v1":{"subscribe":{"payload":{"properties":{"entity-id":{"description":"Account Root Object ID","example":{"entity-id":"00163E5FA59C1ED8B3B95BE1CB795280"},"properties":null,"required":true,"title":"Account Root Object ID","type":"string"},"root-entity-id":{"description":"Account Object ID","example":{"root-entity-id":"00163E5FA59C1ED8B3B9643A4B63D2AF"},"properties":null,"required":true,"title":"Account Object ID","type":"string"}},"required":["entity-id","root-entity-id"],"type":"object"},"summary":"Account.Root.Updated.v1"}}}},"apiSpec":null,"namespace":"c4c-demo","plans":[{"name":"5a70282d-5c83-48b9-9fd2-8b92ab79c199-plan","instanceCreateParameterSchema":null,"displayName":"Default","externalName":"default","namespace":"c4c-demo","relatedServiceClassName":"5a70282d-5c83-48b9-9fd2-8b92ab79c199","__typename":"ServicePlan"}],"__typename":"ServiceClass"}}}
```

Let me know if you require more.

Cheers

Andreas

**Comment by akrausesap** [comment]
@derberg : Can you point me at where the spec is not compliant? Then I will share this with the colleagues

**Comment by derberg** [comment]
In my opinion bug is in the payload spec part of the topics. they define property as `"required": true,` but usually you do it "globally" on the same level as `properties` attribute, as it is also done there `"required": ["entity-id", "root-entity-id"],`

so this one fails:
```

{
				"asyncapi": "1.0.0",
				"info": {
					"description": "SAP Cloud for Customer events",
					"title": "SAP Cloud for Customer events",
					"version": "v1"
				},
				"topics": {
					"Account.Root.Created.v1": {
						"subscribe": {
							"payload": {
								"properties": {
									"entity-id": {
										"description": "Account Root Object ID",
										"example": {
											"entity-id": "00163E5FA59C1ED8B3B95BE1CB795280"
										},
										"properties": null,
										"required": true,
										"title": "Account Root Object ID",
										"type": "string"
									},
									"root-entity-id": {
										"description": "Account Object ID",
										"example": {
											"root-entity-id": "00163E5FA59C1ED8B3B9643A4B63D2AF"
										},
										"properties": null,
										"required": true,
										"title": "Account Object ID",
										"type": "string"
									}
								},
								"required": ["entity-id", "root-entity-id"],
								"type": "object"
							},
							"summary": "Account.Root.Created.v1"
						}
					},
					"Account.Root.Deleted.v1": {
						"subscribe": {
							"payload": {
								"properties": {
									"entity-id": {
										"description": "Account Root Object ID",
										"example": {
											"entity-id": "00163E5FA59C1ED8B3B95BE1CB795280"
										},
										"properties": null,
										"required": true,
										"title": "Account Root Object ID",
										"type": "string"
									},
									"root-entity-id": {
										"description": "Account Object ID",
										"example": {
											"root-entity-id": "00163E5FA59C1ED8B3B9643A4B63D2AF"
										},
										"properties": null,
										"required": true,
										"title": "Account Object ID",
										"type": "string"
									}
								},
								"required": ["entity-id", "root-entity-id"],
								"type": "object"
							},
							"summary": "Account.Root.Deleted.v1"
						}
					},
					"Account.Root.Updated.v1": {
						"subscribe": {
							"payload": {
								"properties": {
									"entity-id": {
										"description": "Account Root Object ID",
										"example": {
											"entity-id": "00163E5FA59C1ED8B3B95BE1CB795280"
										},
										"properties": null,
										"required": true,
										"title": "Account Root Object ID",
										"type": "string"
									},
									"root-entity-id": {
										"description": "Account Object ID",
										"example": {
											"root-entity-id": "00163E5FA59C1ED8B3B9643A4B63D2AF"
										},
										"properties": null,
										"required": true,
										"title": "Account Object ID",
										"type": "string"
									}
								},
								"required": ["entity-id", "root-entity-id"],
								"type": "object"
							},
							"summary": "Account.Root.Updated.v1"
						}
					}
          }
          }

			
```

but this one doesn't:
```

{
				"asyncapi": "1.0.0",
				"info": {
					"description": "SAP Cloud for Customer events",
					"title": "SAP Cloud for Customer events",
					"version": "v1"
				},
				"topics": {
					"Account.Root.Created.v1": {
						"subscribe": {
							"payload": {
								"properties": {
									"entity-id": {
										"description": "Account Root Object ID",
										"example": {
											"entity-id": "00163E5FA59C1ED8B3B95BE1CB795280"
										},
										"properties": null,
										"title": "Account Root Object ID",
										"type": "string"
									},
									"root-entity-id": {
										"description": "Account Object ID",
										"example": {
											"root-entity-id": "00163E5FA59C1ED8B3B9643A4B63D2AF"
										},
										"properties": null,
										"title": "Account Object ID",
										"type": "string"
									}
								},
								"required": ["entity-id", "root-entity-id"],
								"type": "object"
							},
							"summary": "Account.Root.Created.v1"
						}
					},
					"Account.Root.Deleted.v1": {
						"subscribe": {
							"payload": {
								"properties": {
									"entity-id": {
										"description": "Account Root Object ID",
										"example": {
											"entity-id": "00163E5FA59C1ED8B3B95BE1CB795280"
										},
										"properties": null,
										"title": "Account Root Object ID",
										"type": "string"
									},
									"root-entity-id": {
										"description": "Account Object ID",
										"example": {
											"root-entity-id": "00163E5FA59C1ED8B3B9643A4B63D2AF"
										},
										"properties": null,
										"title": "Account Object ID",
										"type": "string"
									}
								},
								"required": ["entity-id", "root-entity-id"],
								"type": "object"
							},
							"summary": "Account.Root.Deleted.v1"
						}
					},
					"Account.Root.Updated.v1": {
						"subscribe": {
							"payload": {
								"properties": {
									"entity-id": {
										"description": "Account Root Object ID",
										"example": {
											"entity-id": "00163E5FA59C1ED8B3B95BE1CB795280"
										},
										"properties": null,
										"title": "Account Root Object ID",
										"type": "string"
									},
									"root-entity-id": {
										"description": "Account Object ID",
										"example": {
											"root-entity-id": "00163E5FA59C1ED8B3B9643A4B63D2AF"
										},
										"properties": null,
										"title": "Account Object ID",
										"type": "string"
									}
								},
								"required": ["entity-id", "root-entity-id"],
								"type": "object"
							},
							"summary": "Account.Root.Updated.v1"
						}
					}
          }
          }

			
```

it is because `required` is expected to be an array of string and not boolean. This is also something that the error in https://www.asyncapi.com/asyncapi-react/ was showing ;)

**Comment by akrausesap** [comment]
@derberg: Well spotted! Removing the required : true fixed the rendering.

**Comment by derberg** [comment]
Awesome. Anyway I keep the bug opened as we should not display an empty `Events` tab but hide it from user and throw proper error in the console for developer

**Comment by derberg** [comment]
Will be fixed once we complete https://github.com/kyma-project/console/issues/684 and https://github.com/kyma-project/kyma/issues/5467 for 1.6 release

**Comment by derberg** [comment]
Closing this one as it will be fixed with 1.6 thanks to asyncapi validation and coversion in the backend
